File: apps/irs-fe/src/types/ticket.ts
export interface IncidentTicket {
  id: string;
  title: string;
  description: string;
  report: string;
  suggestions?: string[];
  severity: 'critical' | 'high' | 'medium' | 'low';
  category: 'kubernetes' | 'infrastructure' | 'ci-cd' | 'other';
  insident_type: 'CPU_HIGH' | 'MEM_HIGH' | 'POD_CRASH' | 'IMAGE_PULL' | 'UNHEALTHY_POD' | 'APP_ERROR' | 'OTHER';
  environment: 'production' | 'staging' | 'development';
  actionStatus: 'auto' | 'manual' | 'pending';
  status: 'open' | 'in-progress' | 'solved' | 'closed' | 'pending';
  reporter: string;
  createdAt: string;
  resolutionTime?: string;
  emailSent: boolean;
  emailSentAt?: string;
  actionTaken?: string;
  affectedServices?: string[];
  tags?: string[];
}

export interface TicketFilters {
  actionStatus?: string;
  severity?: string;
  category?: string;
  environment?: string;
  search?: string;
} 

File: apps/irs-be/internal/dto/ticket.go
package dto

type CreateTicketRequest struct {
	Title            string   `json:"title" validate:"required"`
	Description      string   `json:"description" validate:"required"`
	Severity         string   `json:"severity" validate:"required"`
	Category         string   `json:"category" validate:"required"`
	IncidentType     string   `json:"insident_type" validate:"required"`
	Environment      string   `json:"environment" validate:"required"`
	ActionStatus     string   `json:"actionStatus"`
	Status           string   `json:"status"`
	Reporter         string   `json:"reporter"`
	Suggestions      []string `json:"suggestions,omitempty"`
	AffectedServices []string `json:"affectedServices,omitempty"`
	Tags             []string `json:"tags,omitempty"`
}

type TicketResponse struct {
	ID               string   `json:"id"`
	Title            string   `json:"title"`
	Description      string   `json:"description"`
	Severity         string   `json:"severity"`
	Category         string   `json:"category"`
	IncidentType     string   `json:"insident_type"`
	Environment      string   `json:"environment"`
	ActionStatus     string   `json:"actionStatus"`
	Status           string   `json:"status"`
	Reporter         string   `json:"reporter"`
	CreatedAt        string   `json:"createdAt"`
	ResolutionTime   *string  `json:"resolutionTime,omitempty"`
	EmailSent        bool     `json:"emailSent"`
	EmailSentAt      *string  `json:"emailSentAt,omitempty"`
	ActionTaken      *string  `json:"actionTaken,omitempty"`
	Suggestions      []string `json:"suggestions,omitempty"`
	AffectedServices []string `json:"affectedServices,omitempty"`
	Tags             []string `json:"tags,omitempty"`
}
